Add check for comments that exceed DB col size

salcode opened the pull request:
Add checks for these fields exceeding the DB col size and tests for each
field:
- comment_author
- comment_author_email
- comment_author_url
- comment_content

See #2785

salcode committed:
Add check for comments that exceed DB col size

Add checks for these fields exceeding the DB col size and tests for each
field:

- comment_author
- comment_author_email
- comment_author_url
- comment_content

See #2785

salcode commented:
`CONTRIBUTING.md` asks that I leave a #reviewmerge comment, which I'm not familiar with.  Hopefully, this comment with the hash tag is sufficient.  If not, I'd appreciate any guidance (or a relevant link) on how to add this properly.

Thanks.

rachelbaker commented:
@salcode What do you think about moving this into the `prepare_item_for_database()` method so we are covered on create & update actions?

rachelbaker commented:
This PR would also benefit from a few unit tests.

schlessera commented:
@rachelbaker, the original PR already includes the corresponding unit tests.

salcode committed:
Move DB col size check to prepare_item_for_database()

Move check for comments that exceed DB col size from method
create_item() to prepare_item_for_database().

This allows the check to be performed on both comment create and update
rather than just on create.

See #2785
See #2858

salcode commented:
@rachelbaker thanks for your suggestion.  I've moved the DB column size check to `prepare_item_for_database()` so as to cover both create and update.

Currently, there are unit tests for **create**.  Am I correct you're suggesting we add tests to cover **update**?
